add explicitly - useNativeDriver: false

omerman opened the pull request:
I have seen others proposing this PR.. The difference is that my commit covers all of the places that uses an animation and lacks useNativeDriver.

Closing in favour of #349

omerman committed:
add explicitly - useNativeDriver: false

omerman committed:
Added onBackdrop press.

omerman committed:
onBackdropPress ts

dahveed-rb commented:
@mastermoo bump

gregfenton commented:
@omerman - What is the onBackdropPress for?  I don't see any description for it with the checkins.

omerman commented:
oops, its actually not related to this, I added it as part of my usecase, I wanted to explicitly change a state once someone clicks the backdrop, so I added it.
It is something that I think should be added as well, as a feature, cause by default it calls `reset`
but sometimes I want to change state from my end(the consumer of the library) so onBackdropPress should provide me with more control so I can do it.

lukemcgregor commented on ActionButton.js:
```diff
@@ -241,7 +241,7 @@ const ActionButton = props => {
         activeOpacity={1}
         style={getOverlayStyles()}
-        onPress={reset}
+        onPress={props.onBackdropPress || reset}
```
```suggestion
        onPress={reset}
```

lukemcgregor commented on index.d.ts:
```diff
@@ -28,6 +28,7 @@ export interface ActionButtonProperties extends ViewProperties {
   size?: number,
   autoInactive?: boolean,
   onPress?: () => void,
+  onBackdropPress?: () => void,
```
```suggestion
```

lukemcgregor commented on ActionButton.js:
```diff
@@ -358,7 +358,8 @@ ActionButton.propTypes = {
   accessibilityLabel: PropTypes.string,
-  accessible: PropTypes.bool
+  accessible: PropTypes.bool,
+  onBackdropPress: PropTypes.func,
```
```suggestion
```

omerman commented:
As you can see, opened a different PR containing only the useNative stuff

lukemcgregor commented:
Nah I just proposed some changes and forked you so I had a branch to use in the meantime

omerman commented:
Ok great, I havent noticed. so if my new PR is also not required, you are welcome to close it.
